docgenhtml/plugins/org.polarsys.capella.docgen/src/org/polarsys/capella/docgen/util/CapellaLabelProviderHelper.java

```diff
@@ -0,0 +1,26 @@
+/*******************************************************************************
+ * Copyright (c) 2020 THALES GLOBAL SERVICES.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *  
+ * Contributors:
+ *   Thales - initial API and implementation
+ ******************************************************************************/
+package org.polarsys.capella.docgen.util;
+
+import org.eclipse.emf.common.notify.AdapterFactory;
+import org.eclipse.emf.ecore.EObject;
+import org.eclipse.emf.edit.provider.IItemLabelProvider;
+import org.polarsys.capella.core.model.handler.provider.CapellaAdapterFactoryProvider;
+import org.polarsys.kitalpha.doc.gen.business.core.util.LabelProviderHelper;
+
+public class CapellaLabelProviderHelper extends LabelProviderHelper {
+	
+	public static Object getIItemLabelProvider(EObject eObject) {
```

Comment:
No check on the type of eObject parameter is done. Does the Capella Adapter Factory able to adapt a no Capella element? 

May be you can add a check on the object type, if it is not a capella element, the API LabelProviderHelper.getText() can be used instead. BTW, this API can be used for Capella element and for any element available in the Capella Model.

Reply:
The Capella Adapter Factory is a composed adapter factory ending with an EMF ReflectiveItemProviderAdapterFactory so I would say there is no need for a Check to CapellaElement.